robot_sort/robot_sort.py

```diff
@@ -97,14 +97,117 @@ def sort(self):
+        #There's something inefficient in  here, I fail the second stretch test but pass all others
+        #What if I only did this once my light came on
+        def sort_right(self):
+            while self.can_move_right() == True:
+                self.swap_item()
+                self.move_right()
+                if self.compare_item() > 0:
+                    #Moving this item as far right as I can
+                    while self.compare_item() > 0 and self.can_move_right():
+                        self.move_right()
+                    #Without this if statement, you can never place the highest item at the end
+                    if self.can_move_right():
+                        self.move_left()
+                    self.swap_item()
+                    self.set_light_on()
+                while self.compare_item() != None:
+                    self.move_left()
+                    #Sorting as I go back down to return an item to the None slot
+                    if self.compare_item() == 1:
+                        self.swap_item()
+                self.swap_item()
+                #Back to having no item
+                self.move_right()
+
+        # Sorting down
+        def sort_left(self):
+            while self.can_move_left() == True and self.light_is_on():
+                self.swap_item()
+                self.move_left()
+                if self.compare_item() < 0:
+                    self.swap_item()
+                    self.set_light_on()
+                self.move_right()
+                self.swap_item()
+                #Back to having no item
+                self.move_left()
+
+        #Not sure why this is less efficient than the simple sort_left
+        #Something in here must cause problems if you're already mostly sorted
+        # def sort_left(self):
+        #     while self.can_move_left() == True and self.light_is_on():
+        #         self.swap_item()
+        #         self.move_left()
+        #         if self.compare_item() < 0:
+        #             #Moving this item as far left as I can
+        #             while self.compare_item() < 0 and self.can_move_left():
+        #                 self.move_left()
+        #             if self.can_move_left():
+        #                 self.move_right()
+        #             self.swap_item()
+        #             self.set_light_on()
+        #         while self.compare_item() != None:
+        #             self.move_right()
+        #             #Sorting as I go back up to return an item to the None slot
+        #             if self.compare_item() == -1:
+        #                 self.swap_item()
+        #         self.swap_item()
+        #         #Back to having no item
+        #         self.move_left()
+        
+        self.set_light_on()
+
+        while self.light_is_on() == True:
+            self.set_light_off()
+            sort_right(self)
+            sort_left(self)
+
```

Comment:
I'd be interested to hear how you approached this problem. Your abstractions look really great so it makes your code very readable. Great sprint!